crates/zed/src/languages/scala.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+pub use language::*;
+use lsp::LanguageServerBinary;
+use std::{any::Any, path::PathBuf, str};
+
+pub struct MetalsLspAdapter;
+
+#[async_trait]
+impl LspAdapter for MetalsLspAdapter {
+    fn name(&self) -> LanguageServerName {
+        LanguageServerName("metals".into())
+    }
+
+    fn short_name(&self) -> &'static str {
+        "metals"
+    }
+
+    async fn fetch_latest_server_version(
```

Comment:
Maybe we could fetch the latest version from the official source: `https://scalameta.org/metals/latests.json`. It's a JSON file that contains latest release and snapshot versions. It's automatically updated and used by the Metals webpage/documentation and plugins (Neovim, VS Code). See an example in the nvim-metals: https://github.com/scalameta/nvim-metals/blob/main/lua/metals/install.lua#L138. :slightly_smiling_face: